src/ui/public/courier/fetch/request/request_fetch_params_to_body.js

```diff
@@ -68,6 +69,10 @@ export function requestFetchParamsToBody(
           index = indexList;
         }
 
+
+        $rootScope.filterState = { query: body.query };
```

Comment:
This function is called for each request on the dashboard, which means it gets the specific filters for each panel. We need to find a better way to communicate this information to the values filter.

Reply:
I really don't want to introduce more reliance on rootScope in courier.  I've been trying hard to de-angularize it.  Is there perhaps another way we can achieve the same result? (I ask this without actually looking at the code beyond spying this one tidbit).

Reply:
That's the intention behind the discussion in https://github.com/elastic/kibana/pull/17571#discussion_r180553412